### PR TITLE
[Rule Tuning] Add KEEP Command to all ES|QL Rules

### PR DESCRIPTION
**Pull request by terrancedejesus**
# Pull Request

*Issue link(s)*:

## Summary - What I changed
Adds `keep` ES|QL command to all ES|QL based rules. Add's validation for this as well to enforce it from the ESQLRuleData schema.

## How To Test

Since the validation is in `ESQLRuleData`, running `python -m dev build-release` will load all rules and validate them via their respective data schema.

## Checklist

- [x] Added a label for the type of pr: `bug`, `enhancement`, `schema`, `Rule: New`, `Rule: Deprecation`, `Rule: Tuning`, `Hunt: New`, or `Hunt: Tuning` so guidelines can be generated
- [x] Added the `meta:rapid-merge` label if planning to merge within 24 hours
- [x] Secret and sensitive material has been managed correctly
- [x] Automated testing was updated or added to match the most common scenarios
- [x] Documentation and comments were added for features that require explanation

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/integrations/aws/impact_aws_s3_bucket_enumeration_or_brute_force.toml [hunk 1]
```diff
@@ -89,6 +89,8 @@ type = "esql"
 query = '''
 from logs-aws.cloudtrail*
 | where event.provider == "s3.amazonaws.com" and aws.cloudtrail.error_code == "AccessDenied"
+// keep only relevant fields
+| keep tls.client.server_name, source.address, cloud.account.id
```
Comment (Mikaayenson) [inline rules/integrations/aws/impact_aws_s3_bucket_enumeration_or_brute_force.toml:93]: ++

**Review by Mikaayenson — APPROVED** [review]
lgtm. manually tested w/T online

**Review by w0rk3r — COMMENTED** [review]
How does this affect the telemetry we will get from these rules? What is the motivation behind this change?

**Comment by protectionsmachine** [comment]
## Rule: Tuning - Guidelines

These guidelines serve as a reminder set of considerations when tuning an existing rule.

### Documentation and Context

- [ ] Detailed description of the suggested changes.
- [ ] Provide example JSON data or screenshots.
- [ ] Provide evidence of reducing benign events mistakenly identified as threats (False Positives).
- [ ] Provide evidence of enhancing detection of true threats that were previously missed (False Negatives).
- [ ] Provide evidence of optimizing resource consumption and execution time of detection rules (Performance).
- [ ] Provide evidence of specific environment factors influencing customized rule tuning (Contextual Tuning).
- [ ] Provide evidence of improvements made by modifying sensitivity by changing alert triggering thresholds (Threshold Adjustments).
- [ ] Provide evidence of refining rules to better detect deviations from typical behavior (Behavioral Tuning).
- [ ] Provide evidence of improvements of adjusting rules based on time-based patterns (Temporal Tuning).
- [ ] Provide reasoning of adjusting priority or severity levels of alerts (Severity Tuning).
- [ ] Provide evidence of improving quality integrity of our data used by detection rules (Data Quality).
- [ ] Ensure the tuning includes necessary updates to the release documentation and versioning.

### Rule Metadata Checks

- [ ] `updated_date` matches the date of tuning PR merged.
- [ ] `min_stack_version` should support the widest stack versions.
- [ ] `name` and `description` should be descriptive and not include typos.
- [ ] `query` should be inclusive, not overly exclusive. Review to ensure the original intent of the rule is maintained.

### Testing and Validation

- [ ] Validate that the tuned rule's performance is satisfactory and does not negatively impact the stack.
- [ ] Ensure that the tuned rule has a low false positive rate.

**Review by w0rk3r — APPROVED** [review]

**Comment by imays11** [comment]
@terrancedejesus why are we changing this? When we use `keep` only those fields are included in the alert which makes it hard to triage as none of the other fields are included and a user can't see any other important context outside of the query fields (same issue we have when telemetry filter list only includes fields from rule queries).

**Pull request merged**